lib/pdn/pdn.c

```diff
@@ -347,13 +359,17 @@ int pdn_ctx_create(uint8_t *cid, pdn_event_handler_t cb)
 
 	err = nrf_modem_at_scanf("AT%XNEWCID?", "%%XNEWCID: %d", &ctx_id_tmp);
 	if (err < 0) {
-		pdn_ctx_free(pdn);
+		(void)pdn_ctx_free(pdn);
 		return err;
+	} else if (err == 0) {
+		/* no argument matched */
+		(void)pdn_ctx_free(pdn);
+		return -EFAULT;
```

Comment:
We shouldn't use `EFAULT` here, I suggest something else like `EBADMSG` to indicate that the parsing has failed.